test/unittest/tests/block/test_extent_list_search.cpp

```diff
@@ -0,0 +1,297 @@
+          search_before_after(
+            2 * 4096, &insert_list[1], &insert_list[0]), // Between first and second 8,192.
+          search_before_after(3 * 4096, &insert_list[1], &insert_list[0]), // At second 12,288.
+          search_before_after(
+            4 * 4096, &insert_list[0], &insert_list[2]), // Between second and third 16,384.
+          search_before_after(5 * 4096, &insert_list[0], &insert_list[2]), // At third 20,480.
+          search_before_after(6 * 4096, &insert_list[2], nullptr),         // After third 24,576.
+        };
+
+        /* Setup. */
+        WT_EXTLIST extlist = {};
+
+        /* Insert extents. */
+        for (const off_size &to_insert : insert_list) {
+
+            INFO("Insert: {off " << std::showbase << to_insert.off << ", size " << to_insert.size
+                                 << ", end " << to_insert.end() << "}");
+            REQUIRE(__ut_block_off_insert(session, &extlist, to_insert.off, to_insert.size) == 0);
+        }
+
+        extlist_print_off(extlist);
+
+        /* Test. */
+        WT_EXT dummy;
+        uint32_t idx = 0;
+        for (const search_before_after &expected : expected_before_after) {
+            WT_EXT *before = &dummy;
+            WT_EXT *after = &dummy;
+            /* Call. */
+            __ut_block_off_srch_pair(&extlist, expected.off, &before, &after);
+
+            std::ostringstream line_stream;
+            line_stream << "Verify: " << idx << ". off " << expected.off;
+            if (expected.before != nullptr)
+                line_stream << "; Expected: before: {off " << expected.before->off << ", size "
+                            << expected.before->size << ", end " << expected.before->end() << "}";
+            else
+                line_stream << "; Expected: before == nullptr";
+
+            if (expected.after != nullptr)
+                line_stream << ", after: {off " << expected.after->off << ", size "
+                            << expected.after->size << ", end " << expected.after->end() << "}";
+            else
+                line_stream << ", after == nullptr";
+
+            if (before != nullptr)
+                line_stream << "; Actual: before: {off " << before->off << ", size " << before->size
+                            << ", end " << (before->off + before->size - 1) << "}";
+            else
+                line_stream << "; Actual: before == nullptr";
+
+            if (after != nullptr)
+                line_stream << ", after: {off " << after->off << ", size " << after->size
+                            << ", end " << (after->off + after->size - 1) << "}";
+            else
+                line_stream << ", after == nullptr";
+
+            std::string line = line_stream.str();
+            INFO(line);
+
+            /* Verify. */
+            if (expected.before != nullptr) {
+                REQUIRE(before != nullptr);
+                REQUIRE(before->off == expected.before->off);
```

Comment:
Similarly, you could define `operator==` for `const off_size&`, and then just have something like:
```
                REQUIRE(before != nullptr);
                REQUIRE(*before == *expected.before);
```

Reply:
I would rather not combine multiple tests into one REQUIRE. Keeping the REQUIREs separate is more clear what happened if one of them fails.